src/kaleidoscope/device/ATmega32U4Keyboard.h

```diff
@@ -38,46 +35,57 @@
+  ATMEGA32U4_DEVICE_PROPS_WITH_PROG_KEY(BOARD_, BOOTLOADER_, NAME_,               \
+                                        FORWARD(ROW_PINS_), FORWARD(COL_PINS_),   \
+                                        NO_PROG_KEY)
+
 #define ATMEGA32U4_DEVICE(BOARD_)                                              \
   class BOARD_: public kaleidoscope::device::ATmega32U4Keyboard<BOARD_##Props> {};
 
-#define FORWARD(...) __VA_ARGS__
 
 #ifndef KALEIDOSCOPE_VIRTUAL_BUILD
 
-#define ATMEGA32U4_KEYBOARD(BOARD_, BOOTLOADER_, NAME_, ROW_PINS_, COL_PINS_, PROG_KEY_)  \
-  ATMEGA32U4_DEVICE_PROPS(BOARD_, BOOTLOADER_, NAME_,                                     \
-                          FORWARD(ROW_PINS_), FORWARD(COL_PINS_), FORWARD(PROG_KEY_))     \
+#define ATMEGA32U4_KEYBOARD_WITH_PROG_KEY(BOARD_, BOOTLOADER_, NAME_, ROW_PINS_, COL_PINS_, PROG_KEY_)  \
+  ATMEGA32U4_DEVICE_PROPS_WITH_PROG_KEY(BOARD_, BOOTLOADER_, NAME_,                                     \
+                          FORWARD(ROW_PINS_), FORWARD(COL_PINS_), FORWARD(PROG_KEY_))                   \
   ATMEGA32U4_DEVICE(BOARD_)
 
 #else // ifndef KALEIDOSCOPE_VIRTUAL_BUILD
 
-#define ATMEGA32U4_KEYBOARD(BOARD_, BOOTLOADER_, NAME_, ROW_PINS_, COL_PINS_, PROG_KEY_)  \
-  ATMEGA32U4_DEVICE_PROPS(BOARD_, BOOTLOADER_, NAME_,                                     \
-                          FORWARD(ROW_PINS_), FORWARD(COL_PINS_), FORWARD(PROG_KEY_))     \
-  /* Device definition omitted for virtual device builds.                                 \
-   * We need to forward declare the device name, though, as there are                     \
-   * some legacy extern references to boards whose definition                             \
-   * depends on this.                                                                     \
-   */                                                                                     \
+#define ATMEGA32U4_KEYBOARD_WITH_PROG_KEY(BOARD_, BOOTLOADER_, NAME_, ROW_PINS_, COL_PINS_, PROG_KEY_)  \
+  ATMEGA32U4_DEVICE_PROPS_WITH_PROG_KEY(BOARD_, BOOTLOADER_, NAME_,                                     \
+                          FORWARD(ROW_PINS_), FORWARD(COL_PINS_), FORWARD(PROG_KEY_))                   \
+  /* Device definition omitted for virtual device builds.                                               \
+   * We need to forward declare the device name, though, as there are                                   \
+   * some legacy extern references to boards whose definition                                           \
+   * depends on this.                                                                                   \
+   */                                                                                                   \
   class BOARD_;
 
 #endif // ifndef KALEIDOSCOPE_VIRTUAL_BUILD
 
+#define ATMEGA32U4_KEYBOARD(BOARD_, BOOTLOADER_, NAME_, ROW_PINS_, COL_PINS_)  \
+  ATMEGA32U4_KEYBOARD_WITH_PROG_KEY(BOARD_, BOOTLOADER_, NAME_,                \
```

Comment:
This feels icky.  Every time we add features, we're going to need more and more macros.